code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm

```diff
@@ -185,18 +185,18 @@
 	animate(src, pixel_x = 0, pixel_z = 0, time = 10)
 	sleep(10)
 	playsound(src.loc, 'sound/effects/meteorimpact.ogg', 200, 1)
-	for(var/mob/living/L in range(1,tturf))
-		if(L == src)
-			continue
+	for(var/mob/living/L in orange(1, src))
```

Comment:
`orange` is `range` excluding center atom. And center atom is `src` - dragon wouldn't hurt itself.